pkg/registry/registry/repository/storage/storage.go

```diff
@@ -174,10 +180,59 @@ func (r *REST) Delete(ctx context.Context, name string, deleteValidation rest.Va
 			return nil, false, err
 		}
 	}
+	repoName := fmt.Sprintf("%s/%s", o.Spec.NamespaceName, o.Spec.Name)
+	repoClient, err := client.NewRepository(repoName, o.Spec.TenantID, r.repoEndpoint, r.tokenPrivateKey)
+	if err != nil {
+		return nil, false, err
+	}
+	tags, err := repoClient.ListTag()
+	if err != nil {
+		return nil, false, err
+	}
+	var wg sync.WaitGroup
+	var errsMap sync.Map
+	for _, tag := range tags {
+		wg.Add(1)
+		go deleteTag(tag, repoClient, errsMap)
+	}
+	err = nil
+	for _, tag := range tags {
+		message, ok := errsMap.Load(tag)
+		if !ok {
+			continue
+		}
+		if err == nil {
+			err = fmt.Errorf("failed to delete %s tag %s: %s", repoClient.Name, tag, message.(string))
+		} else {
+			err = fmt.Errorf("%s, failed to delete %s tag %s: %s", err.Error(), repoClient.Name, tag, message.(string))
+		}
+	}
+	if err != nil {
+		return nil, false, err
+	}
+
 	UpdateNamespaceRepoCount(ctx, r.registryClient, o.Spec.NamespaceName, o.Spec.TenantID)
+
 	return r.Store.Delete(ctx, name, deleteValidation, options)
 }
 
+func deleteTag(tag string, rc *client.Repository, errsMap sync.Map) {
+	image := fmt.Sprintf("%s:%s", rc.Name, tag)
+	log.Infof("delete label of image at database: %s", image)
+	if err := rc.DeleteTag(rc.Name, tag); err != nil {
+		log.Errorf("delete repo tag err:%v", err)
+		if regErr, ok := err.(*client.Error); ok {
+			if regErr.Code == http.StatusNotFound {
+				return
+			}
+		}
+		errsMap.Store(tag, err.Error())
+		log.Errorf("failed to delete %s: %v", image, err)
```

Comment:
if it is not 404  error,  line 230 and line 223 are repeated, ingore 404 error is better

Reply:
done